### DISCUSSION TITLE
Running Mint Cinnamon: Fixes in progress.

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1170] [Team Posts]
Hey all. Just checking in.
@kybos I fixed the rack not adjusting when user changes number of displayed rack items.
I recall you mentioning that. Try it out.

I am fixing the mixer - WIP. Particularly the maximizing problem.
The real problem was that this particular wm **ignores** any maximum size that
 has been slapped on a window. I verified this with a 'stock' KDevelop sample Qt app.
I have installed a spacer at the end. Much much better when maximized.
When maximized, still some occasional small artefacts after the last strip when its expander
 handle is moved to the left, no biggy, it erases after a few move movements.
Checking... Pesky to try to get rid of.

Heads up: I noticed that some of the mixer key combos (selecting strips etc.)
 are eaten by the OS, particularly Ctrl Alt Left/Right (reserved for switching workspaces).
In fact, I think several more places may use Alt, never a good idea but what can we do,
 ran out of keys.
So for the mixer at least, I will attempt to create a new shortcut category 'mixer',
 and allow the user to change them, instead of being hard-wired like that.
I will try to change the default from Ctrl Alt Left/Right to just Ctrl Left/Right.
I noticed that there aren't any controls on a strip which really depend on Ctrl Left/Right
 except perhaps the volume display box (standard move among text sections).

Hey @kybos This new theme is looking more sexy all the time.
For the first time I am disliking the light theme in favour of this dark theme.

**Comment by terminator356** [comment]
No more maximize button. Gone.
Nothing worked. Comments from the code:
`      // Due to issues with certain window managers, such as maximize IGNORING`
`      //  our maximum size imposed on the window, maximizing is disabled.`
`      // Several attempts to allow maximizing were unsuccessful.`
`      // The window is full screen width but the scroll area is cut off at our maximum width.`
`      // Seems internally some layer IS respecting our maximum width imposed, cutting it off.`
`      // Tried end spacers, blank widgets autoFillBackground etc. No luck.`
`      // Might be possible but more work needed. Easier to do this:` (Remove the button, that is).

**Comment by kybos** [comment]
> @kybos I fixed the rack not adjusting when user changes number of displayed rack items.
> I recall you mentioning that. Try it out.

Yes, it's working now, thanks!

**Comment by terminator356** [comment]
Hm, I worried about this: I suspected some wms might actually ignore our request
 to remove the maximize button. Sure enough, back in KDE, the maximize button is still there
 and it still works OK.
I suspect perhaps I was supposed to include the Qt::CustomizeWindowHint flag
 when modifying the flags, as per the docs.
But I have not tried that yet, because this is interesting. If there are no complaints
 then maybe we can leave it this way. On KDE or similar it'll just work, and on Gnome
 and similar distros maximizing will be disabled - which is good because it didn't work at least on Mint.

**Comment by spamatica** [comment]
Hey guys, got some mails to read up and respond to.. bear with me..
I just noticed this too, the maximize button is there for me also. 
And it does maximize but behaves a bit wonky, bottom controls are often below the screen and unmaximizing somtimes make the window very small. 
This is kubuntu 18.04 on my Thinkpad t420

**Comment by kybos** [comment]
I noticed some erratic behaviour also on Cinnamon. The mixer changes size in a strange way, usually the window suddenly becomes very small, e.g. when I set focus to one of the strips. Unfortunately it's not always reproducible.
